src/main/java/interface_adapter/club_logged_in/ClubLoggedInState.java

```diff
@@ -1,26 +1,22 @@
 package interface_adapter.club_logged_in;
 
+import entity.data_structure.DataStore;
+import entity.post.Post;
+import entity.user.Student;
+
 /**
  * Club Logged In state.
  */
 public class ClubLoggedInState {
     private String clubName = "";
 
     private String email = "";
-    private String emailError;
 
     private String password = "";
-    private String passwordError;
-
-    public ClubLoggedInState(ClubLoggedInState copy) {
-        this.clubName = copy.clubName;
+    private String errorMessage;
 
-        this.email = copy.email;
-        this.emailError = copy.emailError;
-
-        this.password = copy.password;
-        this.passwordError = copy.passwordError;
-    }
+    private DataStore<Post> posts;
+    private DataStore<Student> members;
```

Comment:
A bit concerned with the existence of an entity in the interface adapter, but I'll ask my instructor tomorrow to check this!

Reply:
Yes, I thought the same. However, my logic behind this decision was that we have no other way of getting the club info without losing too much important data. Moreover, this entity data was passed down from the use case through the output data, so technically there is a whole layer of abstraction (instead of just an interface).

I was wondering, how else would we able to get this data to the view?

Reply:
Hey Roy, I believe Karl's implementation of the state looks okay to me because the state isn't directly modified it's just being used as a placeholder value which is then being accessed later on in the adapter.